tests/test_pep517.py

```diff
@@ -27,11 +27,12 @@ def which(prog: str) -> bool:
         # smoke check for the future if we add another usage
         raise AssertionError(f'Called with {prog}, tests not expecting that usage')
 
+    subprocess_run = subprocess.run
+
     def run(cmd: List[str], *args: object, **kwargs: object) -> subprocess.CompletedProcess:
-        if cmd != ['ninja', '--version']:
-            # smoke check for the future if we add another usage
-            raise AssertionError(f'Called with {cmd}, tests not expecting that usage')
-        return subprocess.CompletedProcess(cmd, 0, f'{ninja}\n', '')
+        if cmd == ['ninja', '--version']:
+            return subprocess.CompletedProcess(cmd, 0, f'{ninja}\n', '')
+        return subprocess_run(cmd, *args, **kwargs)
```

Comment:
The patch to raise `SystemExit` shouldn't be tripping this, right? What's the reasoning behind this change?

Reply:
It actually does. The patch changes a the meson execution from a `subprocess.check_call()` to `subprocess.run()`, which is intercepted by this monkeypatch and everything crumbles.